packages/syft/src/syft/core/node/common/node.py

```diff
@@ -114,25 +120,58 @@ def __init__(
             name=name, network=network, domain=domain, device=device, vm=vm
         )
 
+        # TableBase is the base class from which all ORM classes must inherit
+        # If one isn't provided then we can simply make one.
+        if TableBase is None:
+            TableBase = declarative_base()
+
+        # If not provided a session connecting us to the database, let's just
+        # initialize a database in memory
+        if db_session is None:
+
+            # If a DB engine isn't provided then
+            if engine is None:
+                engine = create_engine("sqlite:///:memory:", echo=False)
+
+            db_session = sessionmaker(bind=engine)()
+
+        # cache these variables on self
+        self.TableBase = TableBase
+        self.engine = engine
+        self.db_session = db_session
+
+        # select which database tables we want to create
+        self.bin_obj_table = BinObject(self.TableBase)
+        self.bin_obj_metadata_table = BinObjectMetadata(self.TableBase)
+
+        # launch the tables in the database
+        self.TableBase.metadata.create_all(engine)
+
+        self.store = BinObjectManager(
+            db_session=self.db_session,
+            bin_obj_table=self.bin_obj_table,
+            bin_obj_metadata_table=self.bin_obj_metadata_table,
+        )
+
         # Any object that needs to be stored on a node is stored here
         # More specifically, all collections of objects are found here
         # There should be NO COLLECTIONS stored as attributes directly
         # on a Node if there is a chance that the collections could
         # become quite numerous (or otherwise fill up RAM).
         # self.store is the elastic memory.
 
-        if db_path is not None:
-            try:
-                self.store = DiskObjectStore(db_path=db_path)
-                log = f"Opened DiskObjectStore at {db_path}."
-                debug(log)
-            except Exception as e:
-                log = f"Failed to open DiskObjectStore at {db_path}. {e}"
-                critical(log)
-        else:
-            self.store = MemoryStore()
-            log = "Created MemoryStore."
-            debug(log)
```

Comment:
We should keep some similar API in which you select the storage backend. More details above.

Reply:
Covered in conversation - SQLAlchemy wrapped by Manager classes will provide this.